configuration_manager/configurations_manager.py

```diff
@@ -0,0 +1,61 @@
+from definitions import CONFIG_FILE
+from xml_parser import Parser
+from config_logger import ConfigLogger
+from directories_manager import DirectoriesManager
+
+
+class ConfigurationsManager(object):
+    """Manages all the configuration settings."""
+    # create output directories
+    dm = DirectoriesManager()
+    __directories = {'output': DirectoriesManager.get_output_directory(),
+                     'logs': DirectoriesManager.get_logs_directory(),
+                     'results': DirectoriesManager.get_results_directory()}
+
+    @classmethod
+    # retrieves directory
+    def get_directory(cls, key):
+        value = cls.__directories.get(key)
+        if value:
+            return cls.__directories.get(key)
+        else:
+            exit(1)  # TODO: better exception handling
```

Comment:
Marking it here for all places of this `# TODO: better exception handling`. Maybe we should collect things likes this in issues. I am sure I will have places with this kind of TODO as well.

Something like:
- an issue with a list of places that need exception handling
- an issue with a list of places that need documentation
- ...

We could then revisit these issues and assign tasks etc.

Reply:
Agreed! a designated issue will be a good place for all such backlogs.